contribs/dvrp/src/main/java/org/matsim/contrib/dvrp/passenger/SinglePassengerPickupActivity.java

```diff
@@ -44,22 +48,22 @@ public SinglePassengerPickupActivity(PassengerHandler passengerHandler, DynAgent
 
 	@Override
 	protected boolean isLastStep(double now) {
-		return passengerAboard && now >= expectedEndTime;
+		return passengersAboard && now >= expectedEndTime;
 	}
 
 	@Override
 	protected void beforeFirstStep(double now) {
-		passengerAboard = passengerHandler.tryPickUpPassenger(this, driver, request.getId(), now);
+		passengersAboard = passengerHandler.tryPickUpPassengers(this, driver, request.getId(), now);
 	}
 
 	@Override
-	public void notifyPassengerIsReadyForDeparture(MobsimPassengerAgent passenger, double now) {
-		if (passenger.getId().equals(request.getPassengerId())) {
+	public void notifyPassengersAreReadyForDeparture(Set<MobsimPassengerAgent> passengers, double now) {
+		if (request.getPassengerIds().containsAll(passengers.stream().map(Identifiable::getId).collect(Collectors.toSet()))) {
```

Comment:
`passengers.stream().map(Identifiable::getId).collect(Collectors.toSet())` is unnecessarily heavy. We can simply collect to list:

`passengers.stream().map(Identifiable::getId).toList()`